teamcity: failed tests on master: test (proposer evaluated kv)/TestParallel, test (proposer evaluated kv)/TestParallel/subquery_retry

The following tests appear to have failed:

[#197834](https://teamcity.cockroachdb.com/viewLog.html?buildId=197834):

```
--- FAIL: test (proposer evaluated kv)/TestParallel (2.040s)
null
--- FAIL: test (proposer evaluated kv)/TestParallel/subquery_retry (2.040s)
logic_test.go:1497: 
	testdata/parallel_test/subquery_retry/final:2: 
	expected:
	    0
	    
	but found (query options: "") :
	    1  
	
logic_test.go:1527: testdata/parallel_test/subquery_retry/final:7: too many errors encountered, skipping the rest of the input
------- Stdout: -------
--- queries start here
--- done: testdata/parallel_test/subquery_retry/setup: 2 tests, 0 failures
--- queries start here
--- queries start here
--- queries start here
--- done: testdata/parallel_test/subquery_retry/txn: 1 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry/txn: 1 tests, 0 failures
--- done: testdata/parallel_test/subquery_retry/txn: 1 tests, 0 failures
--- queries start here
--- done: testdata/parallel_test/subquery_retry/final: 1 tests, 1 failures

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/parallel_test.go
+++ b/pkg/sql/parallel_test.go
@@ -42,7 +42,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/security"
 	"github.com/cockroachdb/cockroach/pkg/sql"
-	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -234,10 +233,6 @@ func (t *parallelTest) setup(spec *parTestSpec) {
 
 func TestParallel(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-
-	if testutils.Stress() {
-		t.Skip()
-	}
 
 	glob := string(*paralleltestdata)
 	paths, err := filepath.Glob(glob)

--- a/pkg/sql/testdata/parallel_test/subquery_retry_multinode/test.yaml
+++ b/pkg/sql/testdata/parallel_test/subquery_retry_multinode/test.yaml
@@ -1,6 +1,3 @@
-# Test is temporarily disabled.
-skip_reason: "#8057"
-
 cluster_size: 5
 
 range_split_size: 32768

--- a/pkg/storage/command_queue.go
+++ b/pkg/storage/command_queue.go
@@ -328,7 +328,7 @@ func (cq *CommandQueue) getWait(
 		//   being consolidated because of range group overlaps.
 		maxWriteTS, minReadTS := hlc.Timestamp{}, hlc.MaxTimestamp
 		cq.oHeap.Init(overlaps)
-		for enclosed := false; cq.oHeap.Len() > 0 && !enclosed; {
+		for cq.oHeap.Len() > 0 {
 			cmd := cq.oHeap.PopOverlap()
 			keyRange := cmd.key
 			cmdHasTimestamp := cmd.timestamp != hlc.Timestamp{}
@@ -388,22 +388,8 @@ func (cq *CommandQueue) getWait(
 					chans = append(chans, cmd.pending)
 				}
 
-				// The current command is a write, so add it to the write RangeGroup and observe if the group grows.
-				if cq.wRg.Add(keyRange) {
-					// We can stop dependency creation early in the case that the write RangeGroup fully encloses
-					// our new range, which means that no new dependencies are needed. This looks only at the
-					// write RangeGroup because even if the combined range group encloses us, there can always be
-					// more reads that are necessary dependencies if they themselves don't overlap any writes. We
-					// only need to perform this check when the write RangeGroup grows.
-					//
-					// We check the write RangeGroup's length before checking if it encloses the new command's
-					// range because we know (based on the fact that these are all overlapping commands) that the
-					// RangeGroup can enclose us only if its length is 1 (meaning all ranges inserted have coalesced).
-					// This guarantees that this enclosure check will always be run in constant time.
-					if cq.wRg.Len() == 1 && cq.wRg.Encloses(newCmdRange) {
-						enclosed = true
-					}
-				}
+				// The current command is a write, so add it to the write RangeGroup.
+				cq.wRg.Add(keyRange)
 
 				// Make sure the current command's range gets added to the combined RangeGroup if we are using it.
 				if overlapRg == cq.rwRg {

--- a/pkg/storage/command_queue_test.go
+++ b/pkg/storage/command_queue_test.go
@@ -468,6 +468,39 @@ func TestCommandQueueTimestamps(t *testing.T) {
 	}
 }
 
+// TestCommandQueueEnclosed verifies that the command queue doesn't
+// fail to return read-only dependencies that a candidate read/write
+// span depends on, despite there being an overlapping span which
+// completely encloses the candidate. See #14434.
+//
+//      Span  TS  RO  a  b  depends
+//         1   2   T  -     n/a
+//         2   3   F  ---   n/a
+// Candidate   1   F  -     spans 1, 2
+func TestCommandQueueEnclosed(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	cq := NewCommandQueue(true)
+
+	spans1 := []roachpb.Span{
+		mkSpan("a", ""),
+	}
+	spans2 := []roachpb.Span{
+		mkSpan("a", "b"),
+	}
+	spansCandidate := []roachpb.Span{
+		mkSpan("a", ""),
+	}
+
+	cmd1 := cq.add(true, makeTS(2, 0), spans1)
+	cmd2 := cq.add(false, makeTS(3, 0), spans2)
+
+	w := cq.getWait(false, makeTS(1, 0), spansCandidate)
+	expW := []<-chan struct{}{cmd2.pending, cmd1.pending}
+	if !reflect.DeepEqual(w, expW) {
+		t.Errorf("expected wait channels %+v; got %+v", expW, w)
+	}
+}
+
 // TestCommandQueueTimestampsEmpty verifies command queue wait
 // behavior when added commands have zero timestamps and when
 // the waiter has a zero timestamp.

